src/redux/books/books.js

```diff
@@ -0,0 +1,55 @@
+import { v4 as uuidv4 } from 'uuid';
+
+const ADDBOOK = 'bookstore/book/ADDBOOK';
+const REMOVEBOOK = 'bookstore/book/REMOVEBOOK'; // name
+
+// const initialState = []; // initial State
+const initialState = [
+  {
+    id: uuidv4(),
+    author: 'john paul',
+    title: 'things fall apart',
+  },
+  {
+    id: uuidv4(),
+    author: 'mercy grace',
+    title: 'orange is the new black',
+  },
+  {
+    id: uuidv4(),
+    author: 'ronalds queen',
+    title: "sorcerer's stone",
+  },
+];
+
+// bookReducer...
+const bookReducer = (state = initialState, action = {}) => {
+  switch (action.type) {
+    case ADDBOOK:
+      return [
+        ...state, {
+          id: uuidv4(),
+          author: action.book.author,
+          title: action.book.title,
+        },
```

Comment:
**OPTIONAL** - Since the new book to be added is an object, I suggest that you do not have to create another object when adding a book, this would help make your code cleaner and easier to read. i.e `return [...state, action.book]`. Please fix this 😄

Reply:
thanks that is actually right.